src/globemap.ts

```diff
@@ -721,33 +719,104 @@ module powerbi.extensibility.visual {
+            if (!tileCacheArray || !tileCacheArray.length)
+                return [];
+
+            tileCacheArray.forEach((obj) => {
+                for (let key in obj) {
+                    if (obj.hasOwnProperty(key)) {
+                        const slpitedArray: string[] = obj[key].split("?")[0].split("/");
+                        let file: string = slpitedArray.pop().split(".")[0];
+                        let domain: string = slpitedArray[2].split(".")[1];
+                        obj[key] = {
+                            file,
+                            domain
                         }
-                        this.localStorageService.setData(GlobeMap.TILE_STORAGE_KEY, tileCache);
-                        this.localStorageService.setData(GlobeMap.TILE_LANGUAGE_CULTURE, this.currentLanguage);
-                        return tileCache;
-                    });
+                    }
+                }
+            });
+
+            return tileCacheArray;
+        }
+
+        private extendBingMetadata(tileCacheArray): any[] {
+            const additionsParams: string = "g=6766&mkt=en-US&shading=hill";
+            const extension: string = "jpeg";
+            const protocol: string = "https://";
+            const firstDomains: string = "tiles.virtualearth.net";
+            const lastDomain: string = "ecn";
+
+            if (!tileCacheArray || !tileCacheArray.length)
+                return null;
+
+            tileCacheArray.forEach((obj) => {
+                for (let key in obj) {
+                    if (obj.hasOwnProperty(key)) {
+                        const mainInfo = obj[key];
+                        obj[key] = `${protocol}${lastDomain}.${mainInfo.domain}.${firstDomains}/tiles/${mainInfo.file}.${extension}?${additionsParams}`;
+                    }
+                }
+            });
+
+            return tileCacheArray;
+        }
+
+        private setBingMetadata(): JQueryPromise<any> {
+            return this.getBingMapsServerMetadata()
+                .then((metadata: BingResourceMetadata) => {
+                    let tileCacheValue = [];
+                    let urlTemplate = metadata.imageUrl.replace("{culture}", this.currentLanguage);
+                    for (let level: number = GlobeMap.initialResolutionLevel; level <= GlobeMap.maxResolutionLevel; ++level) {
+                        let levelTiles = this.generateQuadsByLevel(level, urlTemplate, metadata.imageUrlSubdomains);
+                        this.mapTextures.push(this.createTexture(level, levelTiles));
+                        tileCacheValue.push(levelTiles);
+                    }
+
+                    const minimizedTileCacheData: string = JSON.stringify(this.minimizeBingMetadata(tileCacheValue));
+                    this.localStorageService.set(GlobeMap.TILE_STORAGE_KEY, minimizedTileCacheData);
+                    this.localStorageService.set(GlobeMap.TILE_LANGUAGE_CULTURE, this.currentLanguage);
```

Comment:
Put Culture first here, because it's small value. TILE_STORAGE_KEY might exceed available space.